streams/src/test/java/org/apache/kafka/streams/integration/PauseResumeIntegrationTest.java

```diff
@@ -188,12 +188,11 @@ public void shouldAllowForTopologiesToStartPaused(final boolean stateUpdaterEnab
         kafkaStreams = buildKafkaStreams(OUTPUT_STREAM_1, stateUpdaterEnabled);
         kafkaStreams.pause();
         kafkaStreams.start();
-        waitForApplicationState(singletonList(kafkaStreams), State.RUNNING, STARTUP_TIMEOUT);
+        waitForApplicationState(singletonList(kafkaStreams), State.REBALANCING, STARTUP_TIMEOUT);
```

Comment:
This is mainly for 4) in the description: when we pause the tasks, the state should not transit to RUNNING. This test did not fail because of the bug we are not fixing in line 823 of StoreChangelogReader above.